components/script/dom/htmltextareaelement.rs

```diff
@@ -196,6 +196,16 @@ impl HTMLTextAreaElement {
         HTMLInputElement::directionality_from_value(&value)
     }
 
+    pub(crate) fn move_caret_to_limit(&self) {
+        let UTF8Bytes(content_length) = self.textinput.borrow().len_utf8();
+        let content_length = content_length as u32;
+        self.textinput.borrow_mut().set_selection_range(
+            content_length,
+            content_length,
+            SelectionDirection::None,
+        );
+    }
```

Comment:
Could we move all of the logic to `TextInput` and just call it. That way we can reduce the duplication of the code.
```suggestion
    pub(crate) fn move_caret_to_limit(&self) {
        self.textinput.borrow_mut().move_caret_to_limit();
    }
```

Reply:
Since we use `SetSelectionRange` (as above mentioned), we don't even need this function. Thanks a lot.